compliance/node/index/caching_indexer.go

```diff
@@ -0,0 +1,102 @@
+package index
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/compliance/node"
+	"github.com/stackrox/rox/compliance/utils"
+	v4 "github.com/stackrox/rox/generated/internalapi/scanner/v4"
+)
+
+var (
+	_ node.NodeIndexer = (*cachingNodeIndexer)(nil)
+)
+
+type cachingNodeIndexer struct {
+	indexerConfig *NodeIndexerConfig
+	indexer       node.NodeIndexer
+	cacheDuration time.Duration
+	cachePath     string
+}
+
+// reportWrap is an internal representation of a generated IndexReport with additional metadata
+type reportWrap struct {
+	CacheValidUntil time.Time       // CacheValidUntil indicates the cutoff until the report is considered fresh enough to use.
+	Report          *v4.IndexReport // Report contains the created IndexReport to cache.
+}
+
+// NewCachingNodeIndexer creates a new cached node indexer
+func NewCachingNodeIndexer(indexerConfig *NodeIndexerConfig, cacheDuration time.Duration, cachePath string) node.NodeIndexer {
+	log.Debugf("Creating new cached Node Indexer with cache duration %s and cache path %s",
+		cacheDuration, cachePath)
+	indexer := NewNodeIndexer(indexerConfig)
+	return &cachingNodeIndexer{indexer: indexer, indexerConfig: indexerConfig, cacheDuration: cacheDuration, cachePath: cachePath}
+}
+
+// IndexNode will serve a cached report if it's fresh enough. Otherwise, it creates a new one and caches it.
+func (c cachingNodeIndexer) IndexNode(ctx context.Context) (*v4.IndexReport, error) {
+	r := loadCachedReport(c.cachePath)
+	if r != nil {
+		return r, nil
+	}
+
+	// Cache is too old - create and cache new index report
+	report, err := c.indexer.IndexNode(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "error indexing node")
+	}
+	err = cacheReport(report, c.cachePath, c.cacheDuration)
```

Comment:
I would like a `Debugf` line either before this line or within cacheReport that says we are creating this cache.

Reply:
Done in [77f2490](https://github.com/stackrox/stackrox/pull/13326/commits/77f24902be7ce6a64819af1200e3c8c7badecfa3)